Always show scrollbar regardless of content height

mrtuvn opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->
This PR is improvements for blank/luma theme 

#1 Add some Improvements for blank theme
#2 Default specific width height default for logo via layout but owen can easy override dimensions from backend configuration
#3 Improve magento _reset styles default

### Related Pull Requests
<!-- related pull request placeholder -->

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. Fixes magento/magento2#30705

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
Above screen 768px
1. View category page, contact page, advanced search
2. Logo will not shift from right to left like before when page reload
We should test for both theme luma and blank
but luma has high priority than blank because owner report issue in luma

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->
CC: @krzksz @ptylek @Leland feel free send me any feedback
@gwharton 

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

ihor-sviziev commented:
Hi @mrtuvn,
How this change could be tested?

ihor-sviziev commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run Integration Tests

ihor-sviziev commented on app/design/frontend/Magento/blank/web/css/source/_layout.less:
```diff
@@ -93,7 +93,14 @@
+        .lib-vendor-prefix-flex-basis(auto);
         width: 100%;
+
+        .ie9 & {
+            width: auto;
+        }
```
I believe we don't need to support ie9, so this should be removed.

hostep replied:
Yeah indeed, Magento dropped support for Internet Explorer completely with the release of Magento 2.4.0

Docs: [2.3.x browser support](https://devdocs.magento.com/guides/v2.3/install-gde/system-requirements.html#supported-browsers) vs [2.4.x browser support](https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements.html#supported-browsers)

mrtuvn replied:
I have removed that

ihor-sviziev commented:
Cool! Thank you for the fix!

@magento run all tests

ihor-sviziev commented:
@sidolov could you add some info why Integration tests failing?

sidolov commented:
@magento run all tests

sidolov commented:
@ihor-sviziev I didn't find the cause, let's run them once again

ihor-sviziev commented:
@magentor run Functional Tests B2B, Functional Tests CE, Functional Tests EE

ihor-sviziev reviewed (approved):
✔ Approved.

Failing tests looks not related to changes form this PR.

magento-engcom-team commented:
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8452](https://jira.corp.magento.com/browse/ENGCOM-8452) has been created to process this Pull Request

engcom-Alfa commented:
@magento give me test instance

engcom-Alfa reviewed (changes requested):
Hi @mrtuvn .

It looks like the changes don't solve the problem.

**Actual Result:** :heavy_multiplication_x:  shift from right to left like before when page reload

![Peek 2020-11-12 15-32](https://user-images.githubusercontent.com/51679138/98947923-75b4ac00-24fe-11eb-9baf-6be82ef2b372.gif)
![Peek 2020-11-12 15-35](https://user-images.githubusercontent.com/51679138/98947929-78170600-24fe-11eb-9c66-bc86f5a59b88.gif)

@mrtuvn Could you take a look?
Thank you!

ihor-sviziev commented:
As you can see - it’s caused by the scroll bar that appears on the right
side

On Thu, 12 Nov 2020 at 15:49 Oleksii Stasiv <notifications@github.com>
wrote:

> *@engcom-Alfa* requested changes on this pull request.
>
> Hi @mrtuvn <https://github.com/mrtuvn> .
>
> It looks like the changes don't solve the problem.
>
> *Actual Result:* ✖️ shift from right to left like before when page reload
>
> [image: Peek 2020-11-12 15-32]
> <https://user-images.githubusercontent.com/51679138/98947923-75b4ac00-24fe-11eb-9baf-6be82ef2b372.gif>
> [image: Peek 2020-11-12 15-35]
> <https://user-images.githubusercontent.com/51679138/98947929-78170600-24fe-11eb-9c66-bc86f5a59b88.gif>
>
> @mrtuvn <https://github.com/mrtuvn> Could you take a look?
> Thank you!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/magento/magento2/pull/30714#pullrequestreview-529101918>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAOJOUIGXZR3YJFZVW4N5YLSPPRWTANCNFSM4TECQO2Q>
> .
>

mrtuvn commented:
yes scrollbar appear will make layout move slightly a little bit

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

ihor-sviziev commented on app/design/frontend/Magento/luma/Magento_Theme/layout/default_head_blocks.xml:
```diff
@@ -8,10 +8,5 @@
 <page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
     <head>
         <meta name="viewport" content="width=device-width, initial-scale=1"/>
-        <font src="fonts/opensans/light/opensans-300.woff2"/>
-        <font src="fonts/opensans/regular/opensans-400.woff2"/>
-        <font src="fonts/opensans/semibold/opensans-600.woff2"/>
-        <font src="fonts/opensans/bold/opensans-700.woff2"/>
-        <font src="fonts/Luma-Icons.woff2"/>
```
those included fonts were added here in order to have some font preload that google pagespeed is telling for. Are you sure it's causing  layout shift?

mrtuvn replied:
@ihor-sviziev i'm not sure why this located here. But if you remove it icon still show exactly like before. We can safe remove this

mrtuvn replied:
Previous this from blank theme. After some commit it's moved to luma. Default icon and font imported via less

ihor-sviziev replied:
Yes, they are imported from less files, but it adds font preload, like this:
```html
<link rel="preload" as="font" crossorigin="anonymous" href="https://www.yourdomain.com/static/version1605168925/frontend/Magento/Luma/en_US/fonts/Luma-Icons.woff2" />
```

mrtuvn replied:
I think this is unnecessary. It Take more request to browser

mrtuvn replied:
https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content
Should i need revert this change back to previous ?
Preloading these fonts ensures they are available earlier and are less likely to block the page's render, improving performance

mrtuvn replied:
cc @krzksz what do you think

mrtuvn replied:
Reverted this back to previously

ihor-sviziev commented:
@magento run Functional Tests B2B, Functional Tests EE

ihor-sviziev commented:
@magento run Functional Tests B2B, Functional Tests EE

mrtuvn commented:
rebased + combine commits to one

ihor-sviziev commented:
@magento run all tests

ihor-sviziev commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

ihor-sviziev commented:
@magento run Functional Tests B2B, Functional Tests EE

mrtuvn commented:
Fail tests seem not related with changes

magento-engcom-team commented:
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8452](https://jira.corp.magento.com/browse/ENGCOM-8452) has been created to process this Pull Request

engcom-Alfa commented:
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Deploy a luma or blank vanilla site
2. Expand browser window so there are left and right margins on screen with site centered
3. Create a test product in a test category
4. Browse the following pages, noticing if the position of the logo/navbar/content changes on each page from left/right by a small amount

- Home Page
- Category Page
- Product Page
- Popular Search Terms
- Orders and Returns
- Contact Us
- Advanced Search

**Before:** :heavy_multiplication_x: Sometimes the content shifts right and then back again very quickly, other times it stays offset.

![Peek 2020-11-17 13-09](https://user-images.githubusercontent.com/51679138/99386128-a4ee6300-28da-11eb-98da-316def9a492b.gif)

**After:** :heavy_check_mark: Page Layouts are consistent

**_Luma_**

![Peek 2020-11-17 13-00](https://user-images.githubusercontent.com/51679138/99386287-e5e67780-28da-11eb-8de4-3749665b460a.gif)

**_Blank_**

![Peek 2020-11-17 13-04](https://user-images.githubusercontent.com/51679138/99386310-eed74900-28da-11eb-918c-6d676abd52af.gif)

ihor-sviziev commented:
@engcom-Alfa what about mobile?

mrtuvn commented:
@magento give me test instance

mrtuvn commented:
@ihor-sviziev just do some check in mobile with lighthouse score . Not quite good score in below mobile

mrtuvn commented:
@magento run all tests

engcom-Alfa commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

ihor-sviziev commented:
@magento run Functional Tests B2B, Functional Tests EE, Functional Tests CE, Integration Tests, WebAPI Tests

mrtuvn commented:
Ok it's make sense now after previous day i got so many break in tests report from magento

ihor-sviziev commented:
@magento run Functional Tests B2B, Functional Tests EE

ihor-sviziev commented:
@magento run Integration Tests

mrtuvn commented:
@magento run Functional Tests EE, Functional Tests B2B, Integration Tests

ihor-sviziev commented:
@magento run Functional Tests EE, Functional Tests B2B, Integration Tests

ihor-sviziev commented:
@magento run all tests

ihor-sviziev commented:
I checked - changes still the same, they were just rebased on top of recent 2.4-develop branch, so still approved from my side.

@magento run Functional Tests B2B, Functional Tests EE

mrtuvn commented:
@magento give me 2.4-develop instance

mrtuvn commented:
@magento give me 2.4-develop instance

mrtuvn commented:
@magento run all tests

ihor-sviziev commented:
Changes still the same
@magento run all tests

mrtuvn commented:
rebase code base

mrtuvn commented:
rebase branch PR

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento give me 2.4-develop instance

ihor-sviziev commented:
Hi @mrtuvn,
Could you ping me when you'll finish all the wanted updates and I can start reviewing the pull request?

mrtuvn commented:
yes can we have proceed review on this

ihor-sviziev commented:
@magento run all tests

ihor-sviziev commented:
looks like all the test failures are false positive🤦‍♂️\

@magento run Functional Tests B2B, Functional Tests EE, Integration Tests, Unit Tests, WebAPI Tests

ihor-sviziev commented:
@magento run Functional Tests B2B, Functional Tests EE

ihor-sviziev commented:
@magento run Functional Tests B2B, Functional Tests EE

ihor-sviziev commented:
@magento run Functional Tests B2B, Functional Tests EE

ihor-sviziev commented:
@magento run Functional Tests B2B, Functional Tests EE

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento give me test instance

mrtuvn commented:
@magento give me test instance

mrtuvn commented:
@magento give me test instance

ihor-sviziev commented:
@mrtuvn are you going to change something in your solution?

mrtuvn commented:
@ihor-sviziev there are some problem i need to re-check. Since previous change make some buggy problem while i tested in my local. So i keep this is draft for test

mrtuvn commented:
@magento run all tests

ihor-sviziev reviewed (approved):
✅ Approved
Failing functional tests looks not related to changes from this PR

magento-engcom-team commented:
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8452](https://jira.corp.magento.com/browse/ENGCOM-8452) has been created to process this Pull Request

Leland commented:
> **Fix layout shift while render**

Corrected PR title, IMO:

> **Always show scrollbar regardless of content height**

Can't that say I agree with this PR, or the linked issue, #30705.

This is addressing a non-existent problem, which is, succinctly, that scrollbars show on tall pages but not short ones.

That's just how the web works. If a page is short, it shouldn't show scrollbars. This also is not a case of Cumulative Layout Shift, as the "shift" only happens when navigating between pages - not during the lifecycle of a single page.

mrtuvn commented:
Thank you for some advices and feedbacks

mrtuvn commented:
Do we need this additional ? Added some improvements in magento default _reset.less body tag avoid shifty experience

Leland commented:
My personal position is that that shift is completely normal and does not need addressing.

Forcing scrollbars to always show is, IMO, something most website owners would not want.

mrtuvn commented on lib/web/css/source/lib/_resets.less:
```diff
@@ -12,9 +12,20 @@
+        width: 100%;
     }
 
     article,
```
@Leland do we need this additional ?

mrtuvn committed:
Fix layout shift while render

up

up

update

up

update

Update _layout.less

fix static tests
up

update

update

up

update

up
update


update


update


up


update#3


update


up


update#4

mrtuvn commented:
rebased code and minimal number files changes to avoid conflicts and unintended issues

mrtuvn commented:
@magento give me test instance

mrtuvn commented:
i'm close this solution because this is actually normal problem. That's how browser work
